io_scene_godot/export_godot.py

```diff
@@ -222,6 +222,137 @@ def export_scene(self):
         if in_edit_mode:
             bpy.ops.object.editmode_toggle()
 
+    def export_collection_scene(self, collection):
+        # pylint: disable-msg=too-many-branches
+        """Decide what objects to export, and export them!"""
+        logging.info("Exporting collection: %s of scene %s",
+                     collection.name, self.scene.name)
+
+        in_edit_mode = False
+        if bpy.context.object and bpy.context.object.mode == "EDIT":
+            in_edit_mode = True
+            bpy.ops.object.editmode_toggle()
+
+        # Decide what objects to export
+        # should child collections be exported here as well?
+        for obj in collection.objects:
+            if obj in self.exporting_objects:
+                continue
+            if self.should_export_object(obj):
+                self.exporting_objects.add(obj)
+                # Ensure parents of current valid object is
+                # going to the exporting recursion
+                tmp = obj
+                while tmp is not None:
+                    if tmp not in self.valid_objects:
+                        self.valid_objects.add(tmp)
+                    else:
+                        break
+                    tmp = tmp.parent
+        logging.info("Exporting %d objects", len(self.valid_objects))
+
+        # Scene root
+        root_gd_node = structures.NodeTemplate(
+            collection.name,
+            "Spatial",
+            None
+        )
+        self.escn_file.add_node(root_gd_node)
+        for obj in self.valid_objects:
+            if obj.parent is None:
+                # recursive exporting on root object
+                self.export_object(obj, root_gd_node)
+
+        if "ARMATURE" in self.config['object_types']:
```

Comment:
Similarly with this armature handling: All of the `export_<thingo>` files have this code block. Can it be put in a function?